use index as an arguement to do recursion

whuan20011 opened the pull request.

whuan20011 committed:
use index as an arguement to do recursion

smy20011 commented on numtri1/numtri1.py:
```diff
@@ -0,0 +1,34 @@
+        return dic[cur_index]
+    if cur_index[0] == row - 1:
+        return cur_num
+    cur_num1 = numtri[cur_index[0] + 1][cur_index[1]]
```
Better name is left and right

smy20011 commented on numtri1/numtri1.py:
```diff
@@ -0,0 +1,34 @@
+        return cur_num
+    cur_num1 = numtri[cur_index[0] + 1][cur_index[1]]
+    cur_index1 = (cur_index[0] + 1, cur_index[1])
+    cur_num2 = numtri[cur_index[0] + 1][cur_index[1] + 1]
```
You don't need curr_num here

whuan20011 committed:
numtri
Algorithm: dfs
thinking in a reversed direction
using left and right to represent a node in a tree
watch out the arguments in the function and make it as least as you can
